public/js/event-views.js

```diff
@@ -327,6 +331,103 @@ views.DialogView = Backbone.Marionette.Layout.extend({
                 break;
         }
     },
+
+    submitContactInfo: function(event) {
+        event.preventDefault(); 
+        
+        var preferredContact = {}; //Making an object for preferred contact 
+
+        var noShareVal = false;
+        var scope = $("#my-contact-info-modal");
+        var emailInfo = $("[name=email_info]", scope).val();
+        var twitterHandle = $("[name=twitter_handle]", scope).val();
+        var linkedinURL = $("[name=linkedin_url]", scope).val();
+        var noShareChkBox = $("[name=no-share]", scope).is(':checked');
+
+        //Hiding the error classes
+        $(".empty-contact-info-error", scope).hide();
+
+        $(".email-validate-error").hide();
+        $(".twitter-validate-error").hide();
+        $(".linkedin-validate-error").hide();
+
+        $('#email_info', scope).removeClass('contact_invalid_error');
+        $('#twitter_handle', scope).removeClass('contact_invalid_error');
+        $('#linkedin_url', scope).removeClass('contact_invalid_error');
+
+        if((emailInfo === "" && twitterHandle === "" && linkedinURL === "") && 
+            noShareChkBox === false) {
```

Comment:
``(A && B && C) && D`` is logically equivalent to ``A && B && C && D``.  Losing the parentheses on the first group would improve readability.  If strict equality isn't needed, could also do ``!_.some(emailInfo, twitterHandle, linkedinURL, noShareChkBox)``